server/slack_notifier.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/applikatoni/applikatoni/deploy"
+	"github.com/applikatoni/applikatoni/models"
+
+	"log"
+	"net/http"
+	"text/template"
+)
+
+const slackSuccessSummaryTmplStr = `{{.GitHubRepo}} Successfully Deployed:
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+const slackFailSummaryTmplStr = `:rotating_light:  {{.GitHubRepo}} Deployed Failed :rotating_light: :
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+var slackSuccessSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackSuccessSummaryTmplStr))
+var slackFailSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackFailSummaryTmplStr))
+
+type slackMsg struct {
+	Text string `json:"text"`
+}
+
+func NotifySlack(db *sql.DB, deploymentId int, success bool) {
+	deployment, err := getDeployment(db, deploymentId)
+	if err != nil {
+		log.Printf("Could not find deployment with id %v, %s\n", deploymentId, err)
+		return
+	}
+
+	application, err := findApplication(deployment.ApplicationName)
+	if err != nil {
+		log.Printf("Could not find application with name %v, %s\n", deployment.ApplicationName, err)
+		return
+	}
+
+	target, err := findTarget(application, deployment.TargetName)
+	if err != nil {
+		log.Printf("Could not find target with name %v, %s\n", deployment.TargetName, err)
+		return
+	}
+
+	user, err := getUser(db, deployment.UserId)
+	if err != nil {
+		log.Printf("Could not find User with id %v, %s\n", deployment.UserId, err)
+		return
+	}
+
+	if target.SlackUrl == "" {
+		return
+	}
+
+	var summary string
+	if success {
+		summary, _ = generateSlackSummary(deployment, application, user, slackSuccessSummaryTemplate)
+	} else {
+		summary, _ = generateSlackSummary(deployment, application, user, slackFailSummaryTemplate)
+	}
+	if err != nil {
```

Comment:
Which `err` do you want to check here? You're ignoring the possible error returned by `generateSlackSummary` by assigning it to `_`.

Also, just some advice/idea: I'd assign `template` in the `if/else` branches, instead of summary and then only have one line that calls `generatesSlackSummary`.

```go
var t *template.Template
if success {
   t = slackSuccessSummaryTemplate
} else {
   t = slackFailSummaryTemplate
}

summary, err := generateSlackSummary(deployment, application, user, t)
// [...]
```